pinot-core/src/main/java/com/linkedin/pinot/core/operator/blocks/MatchEntireSegmentDocIdSetBlock.java

```diff
@@ -15,62 +15,26 @@
  */
 package com.linkedin.pinot.core.operator.blocks;
 
-import com.linkedin.pinot.core.common.BlockDocIdSet;
-import com.linkedin.pinot.core.common.BlockDocIdValueSet;
 import com.linkedin.pinot.core.common.BlockId;
-import com.linkedin.pinot.core.common.BlockMetadata;
-import com.linkedin.pinot.core.common.BlockValSet;
-import com.linkedin.pinot.core.common.Predicate;
-import com.linkedin.pinot.core.operator.docidsets.DocIdSetBlock;
+import com.linkedin.pinot.core.operator.docidsets.FilterBlockDocIdSet;
 import com.linkedin.pinot.core.operator.docidsets.SizeBasedDocIdSet;
 
-public class MatchEntireSegmentDocIdSetBlock extends DocIdSetBlock {
+
+public class MatchEntireSegmentDocIdSetBlock extends BaseFilterBlock {
```

Comment:
so when we return this, the explicit cast above (i have added a comment there) will throw exception